### ISSUE TITLE
rest api calls dont work anymore net::ERR_CERT_AUTHORITY_INVALID

### ISSUE THREAD

**Issue by foxzing** [jarrodek/ChromeRestClient#432]
Your description here
The rest api calls that used to work with the old client dont work anymore on the new client.

## Expected outcome
What should happen?
Post Request with Content-Type: application/json
and a json payload should be able to execute a request and be able to return Whether that post was successful or unsuccessful.

## Actual outcome
What happened?
instead is stuck on forever sending that request with the Abort button showing up. Clicking abort just terminates the request.
Saying There was an error during the request.

Request aborted

## Versions
App: 6.1.10.93-stable
Chrome: 49.0.2623.110
Platform: Linux (-)

## Steps to reproduce
1. 
2. 
3.

**Comment by jarrodek** [comment]
Can you please test the beta channel of the app and confirm either issue still exists there or not? Follw the instructions at https://groups.google.com/forum/#!forum/advanced-rest-client-beta-channel to install beta.

**Comment by foxzing** [comment]
I just tried the beta channel version 

The beta version throws an error saying NET::ERR_CERT_AUTHORITY_INVALID. The older client used to allow to open the Rest api url on a new tab and then accept invalid certificate. Since this a web chrome app Opening the url in a chrome browser and accepting this certificate does not give the requisite permission needed to execute this post call

**Comment by jarrodek** [comment]
Would you try this steps http://superuser.com/questions/830506/self-signed-certificate-is-not-appearing-in-chrome-after-importing to install the certificate in your system and tell me if it's working for you? It is working for me but I'd like to be sure.

Basically the transport will check for installed certificates in the system and if found trusted certification authority it will allow the connection and reject it otherwise.

**Comment by foxzing** [comment]
alright will try this, Thank you

**Comment by jarrodek** [comment]
Please let me know if this instructions are working. I tried different ones and my cert is working but this one is much simpler. I'd like to recommend one way of dealing with this issue but not sure which now.

**Comment by jarrodek** [comment]
@foxzing can you confirm if installing the cert on your local machine helps?

**Comment by geeprime** [comment]
Would like to see a feature where it fails with this error, but a "trust this time" checkbox that allows it to proceed without checking the cert authority. Not sure if this would be easier than a separate keystore, as it would be a temporary, in-memory solution (and would need to be re-acknowledged each app relaunch).

**Comment by jarrodek** [comment]
I'm having real trouble with this one.

I don't want to implement cert management in the app because it will require time to code this and later to maintain it. 
All the errors are caused by the lack of certificates installed in the system. Chrome sockets API is not using Chrome settings in that matter. There is a crbug for that but it's unresolved for over a year now. 

My options are:
1. Implement in-app cert management and maintain the code
2. Use XHR/Fetch as a default transport (I have no guarantee that it will work also)
3. Instruct the users on how to install certificate when any of errors above occur 

Installing certs in the system works for me. When I did it I was allowed to make a call. Creating a cert storage in the app is a basically the same as installing it in the system because you need to provide the cert anyway. 
I can add an option to change transport from sockets to XHR or Fetch. But it's a give and take. When switching to regular transport the app will lose ability to set some headers (like user-agent, referer, cookie) and read some headers from the response (like set-cookie).

I've asked a question on stack overflow of [how to implement similar behaviour to accept cert in the app](
http://stackoverflow.com/questions/36552431/creating-an-accept-certificate-functionality-in-chrome-app-like-chrome-does) because I'm not sure how to do it. If anyone have an idea I'm open to suggestions.

**Comment by jarrodek** [comment]
It seems that in Chrome apps XHR will also throw an error when making a request to the server with invalid certificate. Tried it in `xhr` branch with the same result as right now. 
Installation of the certificate might be the only way to accept the certificates.

**Comment by ariskartalos** [comment]
I think that the problem has been generated because new ARC version under Java Script is executed under Chrome engine, that means is a WEB application and falls into WEB application rules and Hosts do not allow accesses on sites with authorizations, CORS support etc... 
Previous version of ARC under Java was executed by JVM, thus it was considered as external Windows application and was treated like this by Hosts.
Maybe I am wrong but give it a thought...

**Comment by jarrodek** [comment]
@ariskartalos Previous version was written in Java but compiled to JavaScript. There's no difference in it. A difference is app's architecture. Old (deprecated apps) worked in chrome. It was something between apps and extensions. New apps (that I was forced to upgrade to) do not use the same settings as Chrome in terms of certificates errors. I tried sockets and xhr. Both failed if the certificate is invalid. 
In other words, if the cert is not installed in Chrome / System (this works for self-signed certificates) or if the certificate is invalid the request will fail. There's no option for apps to ignore certificates errors like chrome does in a tab.

**Comment by ariskartalos** [comment]
I do not have deep knowledge but seems it is big trouble now for you :-(
Is there a way to run older version?
Best regards and keep good work

**Comment by jarrodek** [comment]
There's no point of getting old app because it is deprecated and soon will be removed from Chrome. Chrome Web Store already prohibits uploading old types of apps.

**Comment by jarrodek** [comment]
All, 

Take a look on a discussion I had with chrome team members https://bugs.chromium.org/p/chromium/issues/detail?id=603104. I agree with them that certificates problems should not be resolved with workarounds in apps which will lower the security and is not good practice. 
I spoke to some developers about possibilities and risks I decided not to work on this issue (and similar related to certificates errors). Before I close it I'm open for discussion but web standards are  important for me and the app because they are only guarantee that your API will work in production environment and is future prof.
